dev/doc/proof-engine.txt

```diff
@@ -0,0 +1,126 @@
+Tutorial on the new proof engine for ML tactic writers
+======================================================
+
+Starting from Coq 8.5, a new proof engine has been introduced, replacing the old
+meta-based engine which had a lot of drawbacks. The latter is deprecated and is
```

Comment:
I would be more specific in the "had a lot of drawbacks". As written, I left me wondering what the drawbacks were. I'd either enumerate them, and remove the "a lot", or maybe the more positive phrasing "replacing the old meta-based engine, improving the proof paradigm in these key areas: xxxxx"